### ISSUE TITLE
Move DCV file location to public files directory to avoid intermittent 404s

### ISSUE THREAD

**Issue by richardbporter** [uiowa/uiowa#3806]
The DCV file https://molcellcog.org/.well-known/pki-validation/849A5537A1256BBD44BF06355E091C33.txt was throwing an error intermittently. The error was:
```
php-error web-33203 [14-Jul-2021 08:48:45 America/Chicago] Uncaught PHP Exception Symfony\Component\HttpFoundation\File\Exception\FileNotFoundException: "The file "temporary://849A5537A1256BBD44BF06355E091C33.txt" does not exist" at /vendor/symfony/http-foundation/File/File.php line 36
```

This is because of https://github.com/uiowa/uiowa/issues/3806#issuecomment-880803418. 

## Proposed solution
Save the file in the public files directory but ensure that it gets deleted. These aren't really needed beyond a very short period of time. We'll need to understand file usage to do this correctly.

 - [ ] Determine how long "short period of time" is. Is it under 6 hours where we can rely on automated file cleanup systems?
 - [ ] Determine if a specific filepath is needed, or if we only need to know _what_ the filepath is.
 - [ ] Update https://github.com/uiowa/uiowa/blob/7c806420cfbab6b9876e5d74d4cc85c54e9e8dd2/docroot/modules/custom/sitenow_dcv/src/Form/SitenowDcvFileForm.php#L137 to set a filepath in the public directory, possibly based on the above
 - [ ] Test it and hopefully successfully validate w/the change

**Comment by richardbporter** [comment]
I think this happening inconsistently because it is a temporary file. Refreshing the page many times leads to inconsistent results - some error and some load the file. Some docs suggest we may want to change the temporary file path in out global settings file: https://docs.acquia.com/cloud-platform/manage/files/broken/#setting-the-directory-in-settings-php This might resolve the inconsistencies in redundant hardware setups, i.e. prod.

We might be able to submit the DCV request for now and have it work though @briand44 .

**Comment by briand44** [comment]
I refreshed and it loaded so I submitted DCV. Looks like it worked for now. Thanks!

**Comment by richardbporter** [comment]
>Some docs suggest we may want to change the temporary file path in out global settings file: https://docs.acquia.com/cloud-platform/manage/files/broken/#setting-the-directory-in-settings-php This might resolve the inconsistencies in redundant hardware setups, i.e. prod.

We do not want to do this.

**Comment by GaryRidgway** [comment]
Grooming...
- Current field is just an upload, no delete. we would want to be able to remove it eventually if it was public.
- Options, button and as a part of the process remove the file
- Or mark it as removable for cleanup. Mark file as to be deleted and let it get cleaned up by garbage collector?
    - Currently garbage collection waits 6 hours before marking something as deletable.
    - Could just do this for now and then if we still get the issue, we can change to manual or extend time later?
- As a part of this, we should be testing that this works after being deployed.

- @briand44 , is this supposed to be at a certain specific directory? Say, as part of the public files directory, we are not sure the filepath can stay the same.

**Comment by joewhitsitt** [comment]
Looking at this a bit and reserving some space for @briand44 to fill in, but I think we can specify public in `file_save_upload`'s destination (defaults to temporary) which is causing the issues. If we need to maintain the same `/.well-known/pki-validation/` path, I think that is possible still. Then to deal with cleanup, we could make sure it is added to the file_managed table `setPermanent`? and set the status to 0 so that cron will handle it in the six hours our other files are tied to.

**Comment by briand44** [comment]
Yes, we need to maintain the `/.well-known/pki-validation/` path

**Comment by joewhitsitt** [comment]
@briand44 did this error recently come up to prioritize this issue?

**Comment by briand44** [comment]
Yeah, we had a bunch of domains that I validated a few weeks ago. With more sites moving off of v1 and on to v3 we are seeing this issue more. Most of the validation happens around the same time of the year so we probably won't have many of these to do in the near future and we are able to work around it, it's just a little annoying.

>Determine how long "short period of time" is. Is it under 6 hours where we can rely on automated file cleanup systems?

Yeah, normally the validation happens right away. Because of this issue sometimes it fails initially and continues to retry every 5 minutes I believe. If this issue were resolved, removing the file after 6 hours should be fine.
